src/modules/deal-guarantee/deal-guarantee.controller.ts

```diff
@@ -0,0 +1,66 @@
+import { Body, Controller, Param, ParseArrayPipe, Post } from '@nestjs/common';
+import {
+  ApiBadRequestResponse,
+  ApiBody,
+  ApiCreatedResponse,
+  ApiInternalServerErrorResponse,
+  ApiNotFoundResponse,
+  ApiOperation,
+  ApiParam,
+} from '@nestjs/swagger';
+
+import { DealGuaranteeService } from './deal-guarantee.service';
+import { DealGuaranteeToCreate } from './deal-guarantee-to-create.interface';
+import { CreateDealGuaranteeRequest, CreateDealGuaranteeRequestItem } from './dto/create-deal-guarantee-request.dto';
+import { CreateDealGuaranteeResponse } from './dto/create-deal-guarantee-response.dto';
+
+@Controller()
+export class DealGuaranteeController {
+  constructor(private readonly dealGuaranteeService: DealGuaranteeService) {}
+
+  @Post('deals/:dealIdentifier/guarantees')
+  @ApiOperation({
+    summary: 'Create a new guarantee for a deal.',
+  })
+  @ApiParam({
+    name: 'dealIdentifier',
+    required: true,
+    type: 'string',
+    description: 'The identifier of the deal in ACBS.',
+    example: '00000001',
+  })
+  @ApiBody({
+    type: CreateDealGuaranteeRequestItem,
+    isArray: true,
+  })
+  @ApiCreatedResponse({
+    description: 'The guarantee has been successfully created.',
+    type: CreateDealGuaranteeResponse,
+  })
+  @ApiNotFoundResponse({
+    description: 'The deal was not found.',
+  })
+  @ApiBadRequestResponse({
+    description: 'Bad request.',
+  })
+  @ApiInternalServerErrorResponse({
+    description: 'An internal server error has occurred.',
+  })
+  async createGuaranteeForDeal(
+    @Param('dealIdentifier') dealIdentifier: string,
+    @Body(new ParseArrayPipe({ items: CreateDealGuaranteeRequestItem })) newGuaranteeRequest: CreateDealGuaranteeRequest,
+  ): Promise<CreateDealGuaranteeResponse> {
+    const newGuarantee = newGuaranteeRequest[0];
+    const guaranteeToCreate: DealGuaranteeToCreate = {
```

Comment:
I guess this is to remove unknown fields?

in app.ts we have "whitelist: true," , it should remove unknown fields, but for some reason doesn't work here. Maybe because or array parsing.

Reply:
Global ValidationPipe doesn't apply here. We could use ParseArrayPipe with whitelist: true to remove unknown input fields.

@Body(new ParseArrayPipe({ items: CreateDealGuaranteeRequestItem, whitelist: true })) newGuaranteeRequest: CreateDealGuaranteeRequest,

Reply:
I've left this without the `whitelist` change for now as we're still deciding if we want to keep the 3 class structure (1 for our request, 1 internal interface, and 1 for the ACBS request). If we do keep that then this code is actually to translate between request and internal interface. If we don't keep it then we can refactor this to use the `whitelist` when we flatten the request and internal interface into 1 class